ENGAGEHF/HealthSummary/HealthSummaryView.swift

```diff
@@ -72,6 +118,27 @@ struct HealthSummaryView: View {
             )
         }
     }
+    
+    private func fetchLink() async {
+        do {
+            guard let userId = Auth.auth().currentUser?.uid else {
+                return
+            }
+            let shareHealthSummary = Functions.functions().httpsCallable("shareHealthSummary")
+            let result = try await shareHealthSummary.call([ "userId": userId ] )
+            
+            let dataDictionary = result.data as? [String: Any]
+            self.fullUrl = dataDictionary?["fullUrl"] as? String
+            self.code = dataDictionary?["code"] as? String
```

Comment:
Might be good to kickstart a Task that does a recursive call with a new dispatched `Task` here by calculating the diff between the current time and the expiration time we get from the web service. We need to add some sanity checks that don't call this too often + display a timer in the view.